primedev/windows/libsys.cpp

```diff
@@ -0,0 +1,123 @@
+#include "libsys.h"
+#include "plugins/pluginmanager.h"
+
+#define XINPUT1_3_DLL "XInput1_3.dll"
+
+typedef HMODULE (*WINAPI ILoadLibraryA)(LPCSTR lpLibFileName);
+typedef HMODULE (*WINAPI ILoadLibraryExA)(LPCSTR lpLibFileName, HANDLE hFile, DWORD dwFlags);
+typedef HMODULE (*WINAPI ILoadLibraryW)(LPCWSTR lpLibFileName);
+typedef HMODULE (*WINAPI ILoadLibraryExW)(LPCWSTR lpLibFileName, HANDLE hFile, DWORD dwFlags);
+
+ILoadLibraryA o_LoadLibraryA = nullptr;
+ILoadLibraryExA o_LoadLibraryExA = nullptr;
+ILoadLibraryW o_LoadLibraryW = nullptr;
+ILoadLibraryExW o_LoadLibraryExW = nullptr;
+
+//-----------------------------------------------------------------------------
+// Purpose: Run detour callbacks for given HMODULE
+//-----------------------------------------------------------------------------
+void LibSys_RunModuleCallbacks(HMODULE hModule)
+{
+	if (!hModule)
+	{
+		return;
+	}
+
+	// Get module base name in ASCII as noone wants to deal with unicode
+	CHAR szModuleName[MAX_PATH];
+	GetModuleBaseNameA(GetCurrentProcess(), hModule, szModuleName, MAX_PATH);
```

Comment:
I don't think i tested this when i originally wrote this. Unsure if this is going to work as it doesn't convert from UTF-16 to ASCII. Considering the implementation of `CallLoadLibrary*Callbacks` `GetModuleFileName` might be better suited for the job.